main.go

```diff
@@ -156,8 +172,37 @@ func segment(s *bytes.Buffer) string {
 	return res + "`"
 }
 
+func templateFunctionName(basename string, isPublic bool) string {
+	if isPublic {
+		return basename
+	}
+	return fmt.Sprintf("esc%s", basename)
+}
+
+func header(packageName string, enableExports bool) (string, error) {
```

Comment:
No opposition to using the template package, but I don't think it needs all of this. It should just have the minimum required fields: package name, function prefix (i.e., `esc` or empty string). No need for each thing to have its own property.